homeassistant/components/zwave_js/climate.py

```diff
@@ -470,10 +470,8 @@ async def async_set_temperature(self, **kwargs: Any) -> None:
     async def async_set_hvac_mode(self, hvac_mode: str) -> None:
         """Set new target hvac mode."""
         if not self._current_mode:
-            # Thermostat(valve) with no support for setting a mode
-            raise ValueError(
-                f"Thermostat {self.entity_id} does not support setting a mode"
-            )
+            # Thermostat(valve) has no support for setting a mode, so we make it a no-op
```

Comment:
I think the existing and new solutions are both incorrect.
Just because a device doesn't show an HVAC mode currently, doesn't mean it is not supported. It could be unavailable or unknown at this time as well (just `None`).

I would suggest on checking it against the actual exposed `hvac_modes` instead.

Reply:
`self._current_mode` isn't the actual state of the current HVAC mode, it's the ZwaveValue stored in the zwave_js driver that contains, among other things, the state of the current HVAC mode, like so:
```
{
      "commandClassName": "Thermostat Mode",
      "commandClass": 64,
      "endpoint": 1,
      "property": "mode",
      "propertyName": "mode",
      "metadata": {
        "type": "number",
        "readable": true,
        "writeable": true,
        "min": 0,
        "max": 31,
        "label": "Thermostat mode",
        "states": { "0": "Off", "1": "Heat", "2": "Cool", "3": "Auto" }
      },
      "value": 1
    }
```
If it is `None`, it means that we weren't able to find that ZwaveValue on the device, and we therefore have no way to send a new mode to the device, either because the device doesn't support setting a mode (like in the linked issue) or because the device was not properly interviewed. Because we look for the ZwaveValue on initialization of the entity, I am not sure that this would be correctable by the entity even if the ZwaveValue somehow became available on the driver later.

Does that make sense? And if so, do you still feel that this is an incorrect approach?

Reply:
Maybe we need to remain that variable since as I was writing the above, I realized it captures both the current mode and available modes

Reply:
I changed the logic a bit after discussion with @kpine. He interpreted what you said differently. Whether he got your interpretation correct or not, I think it's better than what I had originally proposed so I changed it accordingly

Reply:
I spoke to kpine on Discord and he interpreted your comment differently. While I am not sure if it addresses your issue, what he proposed does make more sense so I have updated the logic accordingly

Reply:
> self._current_mode isn't the actual state of the current HVAC mode, it's the ZwaveValue stored in the zwave_js driver that contains

That was my confusion spot on. So my comment is invalid. Thanks for pointing that out 👍